static/js/src/advantage/subscribe/checkout/components/UserInfoForm/UserInfoForm.tsx

```diff
@@ -0,0 +1,346 @@
+          document.querySelector("h1")?.scrollIntoView();
+
+          if (error instanceof Error)
+            if (error.message === "email_already_exists") {
+              setError(
+                <>
+                  An Ubuntu Pro account with this email address exists. Please{" "}
+                  <a href="/login">sign in</a> or <a href="/login">register</a>{" "}
+                  with your Ubuntu One account.
+                </>
+              );
+            } else if (error.message.includes("tax_id_invalid")) {
+              setFormikErrors({
+                VATNumber:
+                  "That VAT number is invalid. Check the number and try again.",
+              });
+              setError(
+                <>That VAT number is invalid. Check the number and try again.</>
+              );
+            } else if (error.message.includes("tax_id_cannot_be_validated")) {
+              setFormikErrors({
+                VATNumber:
+                  "VAT number could not be validated at this time, please try again later or contact customer success if the problem persists.",
+              });
+              setError(
+                <>
+                  VAT number could not be validated at this time, please try
+                  again later or contact
+                  <a href="mailto:customersuccess@canonical.com">
+                    customer success
+                  </a>{" "}
+                  if the problem persists.
+                </>
+              );
+            } else {
+              const knownErrorMessage = getErrorMessage({
+                message: "",
+                code: error.message,
+              });
+
+              // Tries to match the error with a known error code and defaults to a generic error if it fails
+              if (knownErrorMessage) {
+                setError(knownErrorMessage);
+              } else {
+                Sentry.captureException(error);
+                setError(
+                  <>
+                    Sorry, there was an unknown error with your credit card.
+                    Check the details and try again. Contact{" "}
+                    <a href="https://ubuntu.com/contact-us">Canonical sales</a>{" "}
+                    if the problem persists.
+                  </>
+                );
+              }
+            }
+        },
+      }
+    );
+  };
+
+  const validateRequired = (value: string) => {
+    let errorMessage;
+    if (!value) {
+      errorMessage = "This field is required.";
+    }
+    return errorMessage;
+  };
+
+  const validateOrganisationName = (value: string) => {
+    let errorMessage;
+    if (!value && values.buyingFor === "organisation") {
+      errorMessage = "This field is required.";
+    }
+    return errorMessage;
+  };
+
+  const displayMode = (
+    <>
+      <Col size={12}>
+        <PaymentMethodSummary />
+      </Col>
+      {values.buyingFor === "organisation" ? (
```

Comment:
On my staging account I get my name as the organisation name, might be a bit unclear.

Reply:
I tested it and this is what I found.

1st the design is this:
![image](https://user-images.githubusercontent.com/6387619/218729486-5c42efaf-0f92-46fa-9e84-5243219e2b81.png)
Organisation name shows as an `h3`. Then your name I think is selected from the SSO data that gets passed in the macaroon.

This is what we have in the new checkout:
![image](https://user-images.githubusercontent.com/6387619/218729821-b365c270-329a-4372-b771-b0cf16f0d9ca.png)

Quite similar to the design.

Not sure why you get your name as an organisation name. Could it be that the organisation name is your name?